metaci/release/tasks.py

```diff
@@ -310,6 +320,120 @@ def all_deps_satisfied(
     )
 
 
+PACKAGE_UPLOAD_TASK_PATHS = [
+    "cumulusci.tasks.create_package_version.CreatePackageVersion",
+    "cumulusci.tasks.salesforce.PackageUpload",
+]
+
+
+def get_package_ids_from_build(build: Build) -> Optional[DependencyGraphItem]:
+    """Traverse the Flow Tasks associated with this Build until we locate
+    a package upload task. Return a DependencyGraphItem with the package
+    details from its output, or None if not found."""
+    for build_flow in build.flows.all():
+        for flow_task in build_flow.tasks.all():
+            if flow_task.class_path in PACKAGE_UPLOAD_TASK_PATHS:
+                # Grab the return values, which will include the 033 and 04t ids
+                all_package_id = flow_task.return_values["package_id"]
+                # The upload tasks for 1GP and 2GP have different return_values shapes
+                all_package_version_id = flow_task.return_values.get(
+                    "version_id"
+                ) or flow_task.return_values.get("subscriber_package_version_id")
+                if all_package_id and all_package_version_id:
+                    return DependencyGraphItem(
+                        AllPackageId=all_package_id,
+                        AllPackageVersionId=all_package_version_id,
+                        Dependencies=[],
+                    )
+
+
+def convert_dependency_graph_to_metapush(rc: ReleaseCohort) -> DependencyGraph:
+    """Convert the Release Cohort's dependency graph (expressed in terms of GitHub URLs)
+    to a MetaPush-style dependency graph (expressed in terms of 04t and 033 package ids)
+    """
+    package_versions_to_github_urls = {}
+    github_urls_to_package_versions = {}
+    dep_graph = DependencyGraph.parse_obj([])
+
+    for release in rc.releases.all():
+        if not release.repo.metapush_enabled:
+            continue
+
+        github_url = release.repo.url
+        # We should have exactly one succeeded Release (Upload Release) build for this Release
+        build = release.build_set.filter(
+            plan__role="release", status=BUILD_STATUSES.success
+        ).first()
```

Comment:
If it should return exactly one, should we use `.get()` in a `try` and catch the `DoesNotExist` error? We aren't protecting here against finding >1 and just taking whichever is found first. Maybe it's adequately enforced elsewhere?